google-cloud-contrib/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageReadChannel.java

```diff
@@ -135,9 +135,33 @@ public int read(ByteBuffer dst) throws IOException {
     }
   }
 
+  private static boolean isReopenable(Throwable exs) {
+    Throwable throwable = exs;
+    while (throwable != null) {
+      if (throwable.getMessage().contains("Connection closed prematurely")
+          || throwable.getCause() instanceof javax.net.ssl.SSLException
```

Comment:
Personal preference only. I switched to imports and factored out getCause().

I also added a loop variant to ensure loop termination.